### PR TITLE
[IMP/FIX] m2o_to_m2m: Use SQL for performance and for avoiding side effects

### PR DESCRIPTION
**Pull request by pedrobaeza**
Same as #234, but for 8.0

### CODE REVIEW

**Comment by legalsylvain** [comment]
Thanks !

**Comment by legalsylvain** [comment]
Oh Wait ! you broke Travis, because of problem of import. You have to import osv at the beginning of the file. 
Could you realize a PR to fix that ?

More generaly, even if the PR for V7 was accepted and so the PR in V8 in trivial, I think that i's not a good thing to very fast merge a PR you wrote. Better to wait at least one :+1:, and wait result of travis.

Regards.

File: openerp/openupgrade/openupgrade.py [hunk 1]
```diff
@@ -537,14 +537,27 @@ def m2o_to_x2m(cr, model, table, field, source_field):
...
                    'field': source_field,
                    })
+    rel, id1, id2 = osv.fields.many2many._sql_names(model._columns[field],
+                                                    model)
     for row in cr.fetchall():
-        model.write(cr, SUPERUSER_ID, row[0], {field: [(4, row[1])]})
+        cr.execute(
+            "INSERT INTO %s (%s, %s) "
+            "VALUES (%s, %s)" %
+            (rel, id1, id2, row[0], row[1]))
```
Comment (hbrunn) [inline openerp/openupgrade/openupgrade.py]: see https://github.com/OpenUpgrade/OpenUpgrade/pull/234/files#r26291892

**Comment by pedrobaeza** [comment]
@hbrunn, Done

@legalsylvain, Travis error is unrelated. I think it can be a new check that has been introduced on flake8, but I haven't seen a lot.

**Comment by legalsylvain** [comment]
openerp/openupgrade/openupgrade.py:541:15: F821 undefined name 'osv'
openerp/openupgrade/openupgrade.py:544:46: F821 undefined name 'osv'
openerp/openupgrade/openupgrade.py:545:15: F821 undefined name 'osv'
openerp/openupgrade/openupgrade.py:554:21: F821 undefined name 'osv'

This is exactly the 4 lines you added, without import of osv.

Please check again, because I think that 7.0 milestone is actually broken.
Regards.

**Comment by pedrobaeza** [comment]
Sorry, my bad in this version. Version 7.0 is not affected, because osv is declared (https://github.com/OpenUpgrade/OpenUpgrade/blob/7.0/openerp/openupgrade/openupgrade.py#L25).

I have fixed this branch, and delete your thumb down to avoid noise when someone oversight the PR.

**Comment by legalsylvain** [comment]
Ow. I understand, you did have the last revision of the 7.0 serie.
OK. Thanks for having cleaned the PR.

**Comment by hbrunn** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
:+1:

**Pull request merged**